_posts/2016-03-16-AngularTS.md

```diff
@@ -0,0 +1,413 @@
+employee.name
+{% endhighlight %}
+
+### Call methods
+{% highlight javascript %}
+employee.eat(eggs);
+{% endhighlight %}
+
+
+## Controllers
+As you know the controller defines the model to the view of your application, methods for every action your require and the scope where you hold a two way binding.
+Because TS offers an object oriented architecture, we can use classes and interfaces instead of functions.
+Interfaces are as in OO languages a contract you make and implementable by classes.
+When implemented, all methods and properties have to be used in the class.
+Classes declare and implement the properties and methods exposed to the view.
+Every class has his own constructor function, in this function we can declare default property values and other initialisation code. 
+
+### Controller Interface
+{% highlight javascript %}
+module Jworks {
+
+    interface IEmployeeController{
+        person:IPerson;
+        save(person:IPerson):void;
+    }
+}
+{% endhighlight %}
+
+The interface will show you the intent of our controller and declare the properties and methods that will be used.
+When you look at the syntax, you see that the properties are strong typed and the type is declared after the colon.
+If you have no aware of which type a property has, you can choose for the general type ‘any’.
+For the methods declared in the interface you have to give the needed parameters and the return type.
+The parameters have the same syntax as the properties declared.  
+
+### Controller Class
+
+{% highlight javascript %}
+module JWorks {
+
+    class EmployeeController implements IEmployeeController{
+
+        static $inject = ["EmployeeService"];
+
+        constructor(private employeeService:IEmployeeService,public employee:IEmployee){
+             this.employee = new Employee(“Nivek”,”Kevin”);
+        }
+
+        save():void{
+             this.employeeService.addEmployee(this.employee).then(function(result){
+             console.log("added successfully!");
+        });
+   }
+
+}
+angular.module("jworks360")
+.controller("EmployeeController", EmployeeController);
+{% endhighlight %}
+
+### Dependency Injection in classes
+When a service is needed in your controller we have to inject these in order to use them.
+In the above example it is important you declare the ‘static $injection’ above your constructor.
+The reason behind this is that in your constructor you will initialise the injected services.
+By doing this the constructor will recognize the injection. 
+If you inject a custom service you have to reference to the related service. 
+
+{% highlight bash %}
+/// <reference path="../services/employee.service.ts"/>
+{% endhighlight %}
+
+
+### Constructor
+TypeScript supports initialisation of your properties and injections in a constructor.
+When declaring properties in your class, you can declare them directly into your constructor.
+As these two examples are right you can have issues in your tests with the second example.
```

Comment:
Would reword to:
**Although **these two examples are **correct** you can have ...